### ISSUE TITLE
Improvement: Add `type` optional param for `create_post(...)` function

### ISSUE THREAD

**Issue by datorr2** [attzonko/mmpy_bot#386]
For example: to create an emote, you would create a post with a `type` of `me`.

![image](https://user-images.githubusercontent.com/28566413/228667956-a5e897db-1b64-4204-8db0-e57d4156257a.png)

**Comment by unode** [comment]
If you want to reply with a reaction you can use `driver.react_to(message, "emoji-name")`.

**Comment by unode** [comment]
Additional mattermost driver actions can be found [in mattermost-autodriver](https://embl-bio-it.github.io/python-mattermost-autodriver/endpoints.html)

**Comment by datorr2** [comment]
> If you want to reply with a reaction you can use `driver.react_to(message, "emoji-name")`.

The use case I was looking to do was not an emoji reaction, but a `/me` emote message.

Like, when the bot first starts up, it could do:

`/me enters stage left.`

And when it shuts down:

`/me exits stage right.`

**Comment by unode** [comment]
I can't find much about this `type` argument.
It seems poorly documented and a little obscure.

I think a pull request adding this feature would be acceptable but I would like to know if this is just for `/me` or other post types are possible?

**Comment by datorr2** [comment]
> I can't find much about this `type` argument. It seems poorly documented and a little obscure.

That is correct.  I haven't been able to find much on it.  I found it when I was trying to figure out how to have the bot send `/me` messages.

> I think a pull request adding this feature would be acceptable but I would like to know if this is just for `/me` or other post types are possible? And if so, can you provide a link to documentation that lists the post types?

No documentation found so far, but there's an entire list of types in the mattermost/mattermost-server project:

https://github.com/mattermost/mattermost-server/blob/c943ed6859b1be013e4da92b1952c607a35258bc/model/post.go

Lines 22 through 53.

**Comment by unode** [comment]
> https://github.com/mattermost/mattermost-server/blob/c943ed6859b1be013e4da92b1952c607a35258bc/model/post.go

This is interesting. #377 might benefit from this as well (`system_join_channel`) even if on the listen side.

If you are able to propose a pull request, I'd happily review.

**Comment by unode** [comment]
I think this could also be a nice feature to showcase in the documentation. Some screenshots of how the different post types look like. It's surprising how many of them there are. I didn't expect the list to be so extensive.
